cmake/HeaderOnlyLibrary.cmake

```diff
@@ -75,7 +75,7 @@ function(header_only_library)
   )
 
   # Add the test, if requested.
-  if (DEFINED ARG_GTEST_SRCS)
+  if (DEFINED ARG_GTEST_SRCS AND ENABLE_TESTING)
```

Comment:
This line has a merge conflict with the new setup on `main`, because we broke this function apart into two separate functions: one for libraries, one for tests.  Accordingly, there's no such thing as `ARG_GTEST_SRCS` anymore.

ericriff/au#1 resolves this merge conflict, by simply adding an early return when this variable is not true.